libsupport/include/katana/ProgressTracer.h

```diff
@@ -157,6 +156,11 @@ class KATANA_EXPORT ProgressTracer {
       const std::string& span_name, std::shared_ptr<ProgressSpan> child_of) = 0;
   ProgressScope SetActiveSpan(std::shared_ptr<ProgressSpan> span);
 
+  /// Close is called when a tracer is finished processing spans
+  /// This should be called to ensure any and all buffered spans are
+  /// flushed
+  virtual void Close() = 0;
```

Comment:
what does it mean to be `virtual` and `private` ?

Reply:
I thought virtual meant it was supposed to be overriden?

Reply:
The unholy trinity of `friend`, `private` and `virtual`.

I now understand what is going on but it was initially confusing. `private` typically means no one except me can call this. `private` and `virtual` is more along the lines of "this is an extension point for subclasses". I normally would make these `protected` but that is a style thing and there is some debate around this: https://stackoverflow.com/questions/2170688/private-virtual-method-in-c

Knowing this, I'd write the function documentation like this:

```cpp
/// Close is called by Finish when ProgressTracer has released all its resources.
///
/// Subclasses override this to release their specific resources.
virtual ProgressTracer::Close() = 0;

/// Close is called by Finish when a ProgressSpan has released all its resources.
///
/// Subclasses override this to release their specific resources.
virtual ProgressSpan::Close() = 0;
```